### PR TITLE
fix: formatting of command sections

### PR DESCRIPTION
**Pull request by a-frantz**
_Describe the problem or feature in addition to a link to the issues._

Two big improvements to formatting of `command-text`.
1. Multi-line placeholders now have an appropriate amount of "temporary indentation" added so they are aligned with the indentation of the open delim.
2. Leading whitespace of command sections is stripped and then normalized to the appropriate level of indentation.

Before submitting this PR, please make sure:

- [x] You have added a few sentences describing the PR here.
- [x] You have added yourself or the appropriate individual as the assignee.
- [x] You have added at least one relevant code reviewer to the PR.
- [x] Your code builds clean without any errors or warnings.
- [x] You have added tests (when appropriate).
- [x] You have updated the README or other documentation to account for these
      changes (when appropriate).
- [x] You have added an entry to the relevant `CHANGELOG.md` (see
      ["keep a changelog"] for more information).
- [x] Your commit messages follow the [conventional commit] style.

[conventional commit]: https://www.conventionalcommits.org/en/v1.0.0/#summary
["keep a changelog"]: https://keepachangelog.com/en/1.0.0/

### CODE REVIEW

File: wdl-format/tests/format/ENCODE-DCC_chip-seq-pipeline/source.formatted.wdl [hunk 1]
```diff
@@ -3336,15 +3328,14 @@ task call_peak {
...
         fi
-
         python3 $(which encode_task_post_call_peak_chip.py) \
             $(ls *Peak.gz) \
             ~{"--ta " + tas[0]} \
             ~{"--regex-bfilt-peak-chr-name '" + regex_bfilt_peak_chr_name + "'"} \
             ~{"--chrsz " + chrsz} \
             ~{"--fraglen " + fraglen} \
             ~{"--peak-type " + peak_type} \
-            ~{"--blacklist " + blacklist}
```
Comment (a-frantz) [inline wdl-format/tests/format/ENCODE-DCC_chip-seq-pipeline/source.formatted.wdl:3347]: I don't know why this trailing space was there, but it's gone now 🤷

File: wdl-format/tests/format/clays_complex_script/source.formatted.wdl [hunk 2]
```diff
@@ -70,7 +70,8 @@ task a_task {
     # the TextMate language that it's a custom struct.
     }
 
-    command <<< >>>
+    command <<<
```
Comment (claymcleod) [inline wdl-format/tests/format/clays_complex_script/source.formatted.wdl:73]: Is this desired? I think, for a totally blank command, I'd prefer `command <<<>>>`. But that is a special case, and I'm also not set on that if you all have made a decision otherwise (please move forward in that case).

Comment (a-frantz) [reply wdl-format/tests/format/clays_complex_script/source.formatted.wdl:73]: @adthrasher has made similar comments about empty sections/tasks/workflows which all currently behave the same (a newline is always inserted after the open delim, even when the body is empty).

IMO these are corner cases not worth the added complexity to address. Granted, that added complexity isn't very great, a few lines in maybe half a dozen places. But that's still extra complexity for an edge case that I just don't think matters.

If folks really want to see empty bodies special cased, I'll add it. But my thinking is we leave as is.

File: wdl-format/tests/format/funky_command_leading_whitespace/source.formatted.wdl [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+task long_leading_whitespace {
+    command <<<
+        echo "hello"
+    >>>
+}
+
+task mixed_leading_whitespace {
+    command <<<
+        echo "hello"
+              echo "world"
```
Comment (adthrasher) [inline wdl-format/tests/format/funky_command_leading_whitespace/source.formatted.wdl:26]: This makes sense based on what we've discussed previously, but I don't particularly like the result.

File: wdl-format/tests/format/clays_complex_script/source.formatted.wdl (hunk 2 above)
Comment (adthrasher) [reply wdl-format/tests/format/clays_complex_script/source.formatted.wdl:73]: I think that's fine, @a-frantz. The additional complication would be, here, this command block isn't truly empty, it contains a single space. Should that be deleted and treated as empty? That would differ from our approach elsewhere.

File: wdl-format/tests/format/ENCODE-DCC_chip-seq-pipeline/source.formatted.wdl
Comment (adthrasher) [inline wdl-format/tests/format/ENCODE-DCC_chip-seq-pipeline/source.formatted.wdl:1]: I wish there was a way in the GitHub diff to set the non-formatted as the original, so you could see the true changes that format made, rather than just the changes from the last commit.

**Review by adthrasher — APPROVED** [review]

File: wdl-format/tests/format/ENCODE-DCC_chip-seq-pipeline/source.formatted.wdl
Comment (a-frantz) [reply wdl-format/tests/format/ENCODE-DCC_chip-seq-pipeline/source.formatted.wdl:1]: That would be nice. Probably the best you can do is to locally checkout the branch and run `diff source.*wdl`, which wouldn't be as nice

File: wdl-format/tests/format/funky_command_leading_whitespace/source.formatted.wdl (hunk 3 above)
Comment (a-frantz) [reply wdl-format/tests/format/funky_command_leading_whitespace/source.formatted.wdl:26]: How would you ideally format this? I think the only way to improve it is to actually parse the Bash to determine that extra whitespace is meaningless. And I've gone on record as saying parsing Bash is a barrel of worms we shouldn't open.
